src/main/java/com/github/manolo8/darkbot/modules/utils/NpcAttacker.java

```diff
@@ -143,9 +145,17 @@ private boolean shouldSab() {
     }
 
     private boolean shouldRsb() {
-        if (!main.config.LOOT.RSB.ENABLED || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
-        if (usedRsb < System.currentTimeMillis() - main.config.LOOT.RSB.AMMO_REFRESH) usedRsb = System.currentTimeMillis();
-
+        if (!main.config.LOOT.RSB.ENABLED || main.config.LOOT.RSB.KEY == null
+                || !target.npcInfo.extra.has(NpcExtra.USE_RSB)) return false;
+        String itemId = main.facadeManager.slotBars.standardBar.slots.stream()
+                .filter(slot -> Objects.equals(main.facadeManager.settings.getCharCode(valueOf
+                        ("SLOTBAR_" + (slot.slotNumber == 10 ? 0 : slot.slotNumber))), main.config.LOOT.RSB.KEY))
```

Comment:
I don't understand how can i reduce complexity of this. Help needed.

Reply:
for each slot in the slotbar you're searching for the key code and then comparing, simply look at my proposed solution below where there's just one after the other, instead of one inside the other